src/sdl/index.ts

```diff
@@ -68,42 +69,47 @@ export class SDL {
     const data = YAML.load(yaml) as v3Sdl;
 
     for (const [name, profile] of Object.entries(data.profiles.compute)) {
-      if (version === "beta3") {
-        SDL.validateGPU(name, profile.resources.gpu);
-      }
-
+      SDL.validateGPU(name, profile.resources.gpu);
       SDL.validateStorage(name, profile.resources.storage);
     }
 
     return data;
   }
 
   static validateGPU(name: string, gpu: v3ResourceGPU | undefined) {
-    if (!gpu) {
-      throw new Error("GPU resource is required for profile " + name);
-    }
+    if (gpu) {
+      if (typeof gpu.units === "undefined") {
+        console.log(JSON.stringify(gpu, null, 2));
+        throw new Error("GPU units must be specified for profile " + name);
+      }
 
-    if (typeof gpu.units === "undefined") {
-      console.log(JSON.stringify(gpu, null, 2));
-      throw new Error("GPU units must be specified for profile " + name);
-    }
+      const units = parseInt(gpu.units.toString());
 
-    const units = parseInt(gpu.units.toString());
+      if (units === 0 && gpu.attributes !== undefined) {
+        throw new Error("GPU must not have attributes if units is 0");
+      }
 
-    if (units == 0 && gpu.attributes !== undefined) {
-      throw new Error("GPU must not have attributes if units is 0");
-    }
+      if (units > 0 && gpu.attributes === undefined) {
+        throw new Error("GPU must not have attributes if units is 0");
+      }
 
-    if (units > 0 && gpu.attributes === undefined) {
-      throw new Error("GPU must not have attributes if units is 0");
-    }
+      if (units > 0 && gpu.attributes?.vendor === undefined) {
+        throw new Error("GPU must specify a vendor if units is not 0");
+      }
 
-    if (units > 0 && gpu.attributes?.vendor === undefined) {
-      throw new Error("GPU must specify a vendor if units is not 0");
-    }
+      if (units > 0 && !GPU_SUPPORTED_VENDORS.some(vendor => !(vendor in (gpu.attributes?.vendor || {})))) {
+        throw new Error(`GPU must be one of the supported vendors (${GPU_SUPPORTED_VENDORS.join(",")}).`);
+      }
 
-    if (units > 0 && gpu.attributes?.vendor?.nvidia === undefined) {
-      throw Error("GPU must specify models if units is not 0");
+      const vendor: string = Object.keys(gpu.attributes?.vendor || {})[0];
+
+      if (units > 0 && gpu.attributes?.vendor[vendor] === undefined) {
+        throw new Error("GPU must specify models if units is not 0");
+      }
```

Comment:
I think the gpu model is optional

Reply:
The test is not necessary now that the test with supported vendors is there